driver-kotlin-sync/src/main/kotlin/com/mongodb/kotlin/client/MongoCollection.kt

```diff
@@ -0,0 +1,1496 @@
+     *   pair.
+     * @param reduceFunction A JavaScript function that "reduces" to a single object all the values associated with a
+     *   particular key.
+     * @param resultClass the target document type of the iterable.
+     * @return an iterable containing the result of the map-reduce operation
+     * @see [map-reduce](https://www.mongodb.com/docs/manual/reference/command/mapReduce/)
+     */
+    @Suppress("DEPRECATION")
+    @Deprecated("Map Reduce has been deprecated. Use Aggregation instead", replaceWith = ReplaceWith(""))
+    public fun <R : Any> mapReduce(
+        mapFunction: String,
+        reduceFunction: String,
+        resultClass: Class<R>
+    ): MapReduceIterable<R> = MapReduceIterable(wrapped.mapReduce(mapFunction, reduceFunction, resultClass))
+
+    /**
+     * Aggregates documents according to the specified map-reduce function.
+     *
+     * @param R the class to decode each resulting document into.
+     * @param clientSession the client session with which to associate this operation
+     * @param mapFunction A JavaScript function that associates or "maps" a value with a key and emits the key and value
+     *   pair.
+     * @param reduceFunction A JavaScript function that "reduces" to a single object all the values associated with a
+     *   particular key.
+     * @param resultClass the target document type of the iterable.
+     * @return an iterable containing the result of the map-reduce operation
+     * @see [map-reduce](https://www.mongodb.com/docs/manual/reference/command/mapReduce/)
+     */
+    @Suppress("DEPRECATION")
+    @Deprecated("Map Reduce has been deprecated. Use Aggregation instead", replaceWith = ReplaceWith(""))
+    public fun <R : Any> mapReduce(
+        clientSession: ClientSession,
+        mapFunction: String,
+        reduceFunction: String,
+        resultClass: Class<R>
+    ): MapReduceIterable<R> =
+        MapReduceIterable(wrapped.mapReduce(clientSession.wrapped, mapFunction, reduceFunction, resultClass))
+
+    /**
+     * Aggregates documents according to the specified map-reduce function.
+     *
+     * @param R the class to decode each resulting document into.
+     * @param mapFunction A JavaScript function that associates or "maps" a value with a key and emits the key and value
+     *   pair.
+     * @param reduceFunction A JavaScript function that "reduces" to a single object all the values associated with a
+     *   particular key.
+     * @return an iterable containing the result of the map-reduce operation
+     * @see [map-reduce](https://www.mongodb.com/docs/manual/reference/command/mapReduce/)
+     */
+    @Suppress("DEPRECATION")
+    @Deprecated("Map Reduce has been deprecated. Use Aggregation instead", replaceWith = ReplaceWith(""))
+    public inline fun <reified R : Any> mapReduce(mapFunction: String, reduceFunction: String): MapReduceIterable<R> =
```

Comment:
Do you really need to add theses deprecated methods ? :)

Reply:
I'd happily not add them :) @jyemin any reasons to include them?

Reply:
I think we can kill them.

Reply:
Done